src/hssm/config.py

```diff
@@ -3,22 +3,33 @@
 # This is necessary to enable forward looking
 from __future__ import annotations
 
+import logging
 from copy import deepcopy
 from dataclasses import dataclass, field
-from typing import TYPE_CHECKING, Any, Literal, Union, cast
+from typing import TYPE_CHECKING, Any, Literal, Type, Union, cast
 
 import bambi as bmb
 
 from .defaults import LogLik, LoglikKind, SupportedModels, default_model_config
+from .param.user_param import (
+    to_dict_shallow,
+)
+from .param.utils import SerializedPrior, deserialize_prior, serialize_prior
 
 if TYPE_CHECKING:
+    from os import PathLike
+
+    import pymc as pm
+    import pytensor
     from pytensor.tensor.random.op import RandomVariable
 
 
 ParamSpec = Union[float, dict[str, Any], bmb.Prior, None]
 
+_logger = logging.getLogger("hssm")
+
 
-@dataclass
+@dataclass(slots=True)
```

Comment:
Yes. This means the dataclass will use the slots implementation rather than a dictionary implementation. In our use case it's faster than the dictionary implementation